wamr/patches/2.4.2/platform_internal.patch

```diff
@@ -0,0 +1,113 @@
+--- a/core/shared/platform/linux/platform_internal.h
++++ b/core/shared/platform/linux/platform_internal.h
+@@ -18,7 +18,7 @@
+ #include <ctype.h>
+ #include <pthread.h>
+ #include <signal.h>
+-#include <semaphore.h>
++// #include <semaphore.h>
+ #include <limits.h>
+ #include <dirent.h>
+ #include <fcntl.h>
+@@ -31,19 +31,74 @@
+ #include <sys/stat.h>
+ #include <sys/mman.h>
+ #include <sys/time.h>
+-#include <sys/timeb.h>
++// #include <sys/timeb.h>
+ #include <sys/uio.h>
+ #include <sys/ioctl.h>
+ #include <sys/socket.h>
+ #include <sys/resource.h>
++#include <sys/threads.h> // Header file from libphoenix
++#include <arpa/inet.h> // Header file from libphoenix
+ 
+ #ifdef __cplusplus
+ extern "C" {
+ #endif
+ 
+-#ifndef BH_PLATFORM_LINUX
+-#define BH_PLATFORM_LINUX
+-#endif
++#ifndef BH_PLATFORM_PHOENIX
++#define BH_PLATFORM_PHOENIX
++#endif
++
++/* Config definitions */
++#define DISABLE_CLOCK_NANOSLEEP
++
++/* Missing definitions */
++double nearbyint(double x);
++float nearbyintf(float x);
++#define O_ACCMODE 3 // Mask for accessing R/W access mode
```

Comment:
```suggestion
```

`O_ACCMODE` is already defined in `fcntl.h`. Moreover its value here is incorrect.
```c
#define O_RDONLY   0x00001U
#define O_WRONLY   0x00002U
#define O_RDWR     0x00004U
#define O_APPEND   0x00008U
#define O_CREAT    0x00100U
#define O_TRUNC    0x00200U
#define O_EXCL     0x00400U
#define O_SYNC     0x00800U
#define O_NONBLOCK 0x01000U
#define O_NDELAY   O_NONBLOCK
#define O_NOCTTY   0x02000U
#define O_CLOEXEC  0x04000U
#define O_RSYNC    0x08000U
#define O_DSYNC    0x10000U

#define O_ACCMODE (O_RDONLY | O_WRONLY | O_RDWR)
```

That'd make `O_ACCMODE` resolve to 7, not 3